air/src/execution_step/instructions/canon_stream_map_scalar.rs

```diff
@@ -70,12 +71,21 @@ fn epilog_closure<'closure, 'name: 'closure>(scalar_name: &'name str) -> Box<Can
               exec_ctx: &mut ExecutionCtx<'_>,
               trace_ctx: &mut TraceHandler|
               -> ExecutionResult<()> {
-            let value = JValuable::as_jvalue(&&canon_stream).into_owned();
-            let tetraplet = canon_stream.tetraplet();
-            let position = trace_ctx.trace_pos().map_err(UncatchableError::from)?;
+            use crate::CanonStreamMapError::NoDataToProduceScalar;
+
+            let tetraplet = canon_stream.tetraplet().clone();
             let peer_pk = tetraplet.peer_pk.as_str().into();
 
-            let value = CanonResultAggregate::new(Rc::new(value), peer_pk, &tetraplet.json_path, position);
+            let value = canon_stream
+                .get_values()
+                .first()
```

Comment:
It worth commenting such a hack; moreover, (debug_)assert that the element is single.